library/modules/Units.cpp

```diff
@@ -644,10 +644,9 @@ bool Units::isDanger(df::unit *unit) {
 
     return isCrazed(unit)
         || isInvader(unit)
-        || isOpposedToLife(unit)
         || isAgitated(unit)
         || isSemiMegabeast(unit)
-        || isNightCreature(unit)
+        || ((isOpposedToLife(unit) || isNightCreature(unit)) && !unit->flags2.bits.visitor)
```

Comment:
shouldn't this also check `visitor_uninvited`? If they *are* an unwanted visitor, shouldn't this return true?

maybe that can be an independent check further up as one of the OR clauses.